provider/pkg/provider/provider.go

```diff
@@ -374,13 +374,15 @@ func (k *kubeProvider) DiffConfig(ctx context.Context, req *pulumirpc.DiffReques
 	}
 
 	// We can't tell for sure if a computed value has changed, so we make the conservative choice
-	// and force a replacement.
-	if news["kubeconfig"].IsComputed() {
-		return &pulumirpc.DiffResponse{
-			Changes:  pulumirpc.DiffResponse_DIFF_SOME,
-			Diffs:    []string{"kubeconfig"},
-			Replaces: []string{"kubeconfig"},
-		}, nil
+	// and force a replacement. Note that getActiveClusterFromConfig relies on all three of the below properties.
+	for _, key := range []resource.PropertyKey{"kubeconfig", "context", "cluster"} {
+		if news[key].IsComputed() {
+			return &pulumirpc.DiffResponse{
+				Changes:  pulumirpc.DiffResponse_DIFF_SOME,
+				Diffs:    []string{string(key)},
+				Replaces: []string{string(key)},
+			}, nil
+		}
```

Comment:
This fix is to avoid a panic that otherwise occurs in `getActiveClusterFromConfig` if either `context` or `cluster` has a computed value. I think it is safe to be 'more conservative' during preview since it is not determinative as to whether a resource would actually be replaced.